### PR TITLE
ISEScan tool for inserted element detection in prokaryote

### PR DESCRIPTION
**Pull request by pimarin**
FOR CONTRIBUTOR:
* [X] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [ ] - License permits unrestricted use (educational + commercial)
* [X] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

**Review by bgruening — COMMENTED** [review]
@pimarin in general this tools looks great to me. I'm just worried about the tests, the outputs files are all empty. Are the tests really working?

**Comment by pimarin** [comment]
Hum sure not !
This a problem of output direction, it seem to not know where are the output files after running.

**Comment by pimarin** [comment]
In the logfile I have it :

` Write 2 IS copies to results/0/dataset_5d059bdd-e417-47ca-bbe9-298ea6696953.dat.csv and 
results/0/dataset_5d059bdd-e417-47ca-bbe9-298ea6696953.dat.tsv and results/0/dataset_5d059bdd-e417-47ca-bbe9-298ea6696953.dat.raw
Write summarized result to results/0/dataset_5d059bdd-e417-47ca-bbe9-298ea6696953.dat.sum`
So i have a random number between the directory `results` and the `dataset`

**Comment by bernt-matthias** [comment]
I guess the outputs are simply written to the prefix `output` (which might be a directory or just a file name prefix). You can either move the created files to the output cheetah variables, e.g. `mv outputsXXX '$summary'` or you use the `from_workdir` attribute of the output in order to tell Galaxy where to get the files. 

Also the tool seems to write files next to the input files (eg https://github.com/xiezhq/ISEScan/blob/110216cfa91e382322cd92e3e13ccb5eac7753a0/isescan.py#L21). This should fail here (because we use docker containers for testing) .. but may be successful in local tests with conda. I guess you will need to symlink (or copy) the inputs to the job working dir.

**Comment by pimarin** [comment]
> I guess the outputs are simply written to the prefix `output` (which might be a directory or just a file name prefix). You can either move the created files to the output cheetah variables, e.g. `mv outputsXXX '$summary'` or you use the `from_workdir` attribute of the output in order to tell Galaxy where to get the files.
> 
> Also the tool seems to write files next to the input files (eg https://github.com/xiezhq/ISEScan/blob/110216cfa91e382322cd92e3e13ccb5eac7753a0/isescan.py#L21). This should fail here (because we use docker containers for testing) .. but may be successful in local tests with conda. I guess you will need to symlink (or copy) the inputs to the job working dir.

I already tried the `from_workdir` it failed, but the output option is to create a folder with output files, I tried to use symlink and a mv option.
Thanks !

**Comment by pimarin** [comment]
I failed to try it, 

```
mv 'results/**/*.sum' $summary
        #if '$output_format' == 'raw'
            mv 'results/**/*.fna.raw' $is_results &&
        #else if '$output_format' == 'csv'
            mv 'results/**/*.fna.csv' $is_results &&
        #else if '$output_format' == 'tsv'
            mv 'results/**/*.fna.tsv' $is_results &&
        #end if
        mv 'results/**/*.fna.gff' '$gff_file' &&
        mv 'results/**/*.fna.is.fna' '$is_fasta' &&
        mv 'results/**/*.fna.orf.faa' '$tpase_amino' &&
        mv 'results/**/*.fna.orf.fna' '$tpase_fasta' &&
        mv 'results/**/*.log' '$logfile'
```
don't moove files

**Comment by pimarin** [comment]
maybe a discover dataset ?

**Comment by bernt-matthias** [comment]
> maybe a discover dataset ?

Probably .. maybe also each of the outputs probably should be a collection.

So the result directory looks like:

```
results
├── 2
│   ├── dataset_9e23156d-78cc-4652-9b90-3b80d4c981cc.dat.csv
│   ├── dataset_9e23156d-78cc-4652-9b90-3b80d4c981cc.dat.gff
│   ├── dataset_9e23156d-78cc-4652-9b90-3b80d4c981cc.dat.is.fna
│   ├── dataset_9e23156d-78cc-4652-9b90-3b80d4c981cc.dat.orf.faa
│   ├── dataset_9e23156d-78cc-4652-9b90-3b80d4c981cc.dat.orf.fna
│   ├── dataset_9e23156d-78cc-4652-9b90-3b80d4c981cc.dat.out
│   ├── dataset_9e23156d-78cc-4652-9b90-3b80d4c981cc.dat.raw
│   ├── dataset_9e23156d-78cc-4652-9b90-3b80d4c981cc.dat.sum
│   └── dataset_9e23156d-78cc-4652-9b90-3b80d4c981cc.dat.tsv
├── hmm
│   └── 2
│       ├── clusters.faa.hmm.dataset_9e23156d-78cc-4652-9b90-3b80d4c981cc.dat.faa
│       └── clusters.single.faa.dataset_9e23156d-78cc-4652-9b90-3b80d4c981cc.dat.faa
└── proteome
    └── 2
        ├── dataset_9e23156d-78cc-4652-9b90-3b80d4c981cc.dat.faa
        ├── dataset_9e23156d-78cc-4652-9b90-3b80d4c981cc.dat.ffn
        ├── dataset_9e23156d-78cc-4652-9b90-3b80d4c981cc.dat.gff
        └── dataset_9e23156d-78cc-4652-9b90-3b80d4c981cc.dat.out
```

Can the `hmm` and `proteome` directories be ignored?

Can there be multiple supdirs, e.g., `1`, `2`, ...?

**Comment by pimarin** [comment]
Finally, with a symbolic link, I retrieve all files, maybe more simple than the collection ?

File: tools/isescan/isescan.xml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+            | tee '$logfile'
+        #end if
+    ]]>
+    </command>
+    <inputs>
+        <param name="input_file" type="data" format="fasta" label="Genome fasta input" help="Fasta sequence to analyse for IS"/>
+        <param name="remove_short_is" argument="--removeShortIS" type="boolean" truevalue="--removeShortIS" falsevalue=""
+               label="Remove incomplete IS elements"
+               help="Remove partial IS elements which include IS element with length &gt; 400 or single copy IS element without perfect TIR"/>
+        <param name="log_activate" type="boolean" truevalue="true" falsevalue="false" checked="True" label="Keep the log file"/>
```
Comment (bgruening) [inline tools/isescan/isescan.xml:26]: you probably wanted to filter the logfile in the outputs with this param?

File: tools/isescan/isescan.xml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+            <param name="remove_short_is" value="true"/>
+            <output name="logfile" value="test_2.log" lines_diff="50"/>
+            <output name="summary" value="test_2.fna.sum" lines_diff="1"/>
+            <output name="all_results" value="test_2.fna.tsv" lines_diff="1"/>
+            <output name="annotation" value="test_2.fna.gff" lines_diff="1"/>
+            <output name="is_fasta" value="test_2.fna.is.fna" lines_diff="1"/>
+            <output name="orf_fna" value="test_2.fna.orf.fna" lines_diff="1"/>
+            <output name="orf_faa" value="test_2.fna.orf.faa" lines_diff="1"/>
+        </test>
+        <test expect_num_outputs="7"> <!-- TEST_1 remove short is and no logfile -->
```
Comment (bgruening) [inline tools/isescan/isescan.xml:59]: this should be 6 when the log file is not wanted.

**Review by bgruening — APPROVED** [review]

**Comment by bgruening** [comment]
Very clean and nice! Thanks @pimarin

**Pull request merged**